### PR TITLE
[WIP] Add mypy syntax cheat sheet.

### PR DESCRIPTION
**Pull request by timabbott**
Here's a first version of a mypy syntax cheat sheet, intended to be linked to from the CI of project using mypy ; feedback (or picking this up and improving it directly during the Pycon sprints) is welcome!

### CODE REVIEW

**Comment by gvanrossum** [comment]
Thanks! Looks like you accidentally added a non-global branch of typeshed to the diff.

File: docs/source/cheat_sheet.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   x = set([6, 7]) # type: Set[int]
+   # For mappings, we need the types of both keys and values
+   x = dict(field=2.0) # type: Dict[str, float]
+   # For tuples, we specify the types of all the elements
+   x = (3, "yes", 7.5) # type: Tuple[int, str, float]
+
+   # six.text_type is str/unicode in Python 2 and string but not bytes in Python 3
+   x = ["string", u"unicode"] # type: List[text_type]
+   # If something could be one of a few types, use Union
+   x = [3, 5, "test", "fun"] # type: Union[int, str]
```
Comment (JukkaL) [inline docs/source/cheat_sheet.rst]: Annotation should be `List[union[int, str]]`.

File: docs/source/cheat_sheet.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   x = re.match(r'[0-9]+', "15") # type: Match
```
Comment (JukkaL) [inline docs/source/cheat_sheet.rst]: A more precise type would be `Match[str]`.

**Comment by gvanrossum** [comment]
Assuming Tim's busy I can manually commit this and fix it up.

**Comment by timabbott** [comment]
Yep am busy, feel free to take this over!

**Comment by gvanrossum** [comment]
Committed this with Jukka's two fixes. It's not referenced from any other docs yet, I'll try to find a place for that.

**Pull request closed without merging**